### PR TITLE
Suppress excess errors in kubectl logs due to Btree

### PR DESCRIPTION
**Pull request by ydb242**
# Problem Statement
We observe a lot of expected ERRORs in kubectl logs. These errors make reading the logs a bit problematic, and might hide the errors. 
 
# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: cas/ctg_store.c [hunk 1]
```diff
@@ -1274,7 +1274,7 @@ static int ctg_op_exec(struct m0_ctg_op *ctg_op, int next_phase)
 		m0_be_op_done(beop);
 		break;
 	}
-	ctg_op->co_rc = rc;
+	ctg_op->co_rc = M0_RC(rc);
```
Comment (kanchan-chaudhari) [inline cas/ctg_store.c:1277]: Do we really need this Log?? This function might be getting called manier times, I think we can skip this log.

File: cob/cob.c [hunk 2]
```diff
@@ -1022,8 +1022,9 @@ M0_INTERNAL int m0_cob_domain_mkfs(struct m0_cob_domain *dom,
 
 	m0_buf_init(&key, &omgkey, sizeof omgkey);
 	m0_buf_init(&rec, &omgrec, sizeof omgrec);
-	cob_table_insert(dom->cd_fileattr_omg, tx, &key, &rec);
-
+	rc = cob_table_insert(dom->cd_fileattr_omg, tx, &key, &rec);
+	if (rc != 0)
+		return M0_RC(rc);
```
Comment (shashank-parulekar) [inline cob/cob.c:1027]:
```suggestion
		return M0_ERR(rc);
```

File: cob/cob.c [hunk 3]
```diff
@@ -2049,8 +2050,9 @@ M0_INTERNAL int m0_cob_name_update(struct m0_cob *cob,
 
 	m0_buf_init(&key, tgtkey, m0_cob_nskey_size(tgtkey));
 	/* here @val consists value to insert */
-	cob_table_insert(cob->co_dom->cd_namespace, tx, &key, &val);
-
+	rc = cob_table_insert(cob->co_dom->cd_namespace, tx, &key, &val);
+	if (rc != 0)
+		return M0_RC(rc);
```
Comment (shashank-parulekar) [inline cob/cob.c:2055]:
```suggestion
		return M0_ERR(rc);
```

File: cob/cob.c [hunk 4]
```diff
@@ -2210,7 +2213,9 @@ M0_INTERNAL int m0_cob_ea_set(struct m0_cob *cob,
 
 	m0_buf_init(&key, eakey, m0_cob_eakey_size(eakey));
 	m0_buf_init(&val, earec, m0_cob_earec_size(earec));
-	cob_table_insert(cob->co_dom->cd_fileattr_ea, tx, &key, &val);
+	rc = cob_table_insert(cob->co_dom->cd_fileattr_ea, tx, &key, &val);
+	if (rc != 0)
+		return M0_RC(rc);
```
Comment (shashank-parulekar) [inline cob/cob.c:2218]:
```suggestion
		return M0_ERR(rc);
```

**Review by shashank-parulekar — CHANGES_REQUESTED** [review]

File: be/extmap.c [hunk 5]
```diff
@@ -579,6 +579,8 @@ M0_INTERNAL void m0_be_emap_merge(struct m0_be_emap_cursor *it,
 
 	m0_rwlock_write_lock(emap_rwlock(it->ec_map));
 	rc = emap_it_pack(it, be_emap_delete_wrapper, tx);
+	if (rc != 0)
+		M0_RC(rc);
```
Comment (shashank-parulekar) [inline be/extmap.c:583]:
```suggestion
		M0_ERR(rc);
```

**Review by shashank-parulekar — CHANGES_REQUESTED** [review]

File: cas/ctg_store.c (hunk 1 above)
Comment (kanchan-chaudhari) [reply cas/ctg_store.c:1277]: @shashank-parulekar  can you confirm this comment??

File: cas/ctg_store.c [hunk 6]
```diff
@@ -2107,7 +2107,7 @@ M0_INTERNAL int m0_ctg_ctidx_delete_sync(const struct m0_cas_id *cid,
 							   &kv_op, tx));
 
 	m0_chan_broadcast_lock(&ctidx->cc_chan.bch_chan);
-	return rc;
+	return M0_RC(rc);
```
[the remark this answers is not in the document]
Comment (shashank-parulekar) [reply cas/ctg_store.c:2110]: Use M0_ERR if (rc != 0). If rc == 0 then no need to use M0_RC() also.. Same change will be needed in line 2087 above.

File: cas/ctg_store.c (hunk 1 above)
Comment (shashank-parulekar) [reply cas/ctg_store.c:1277]: Yes this log is needed only if (rc != 0) and then also you should use M0_ERR() and not M0_RC().

**Review by shashank-parulekar — CHANGES_REQUESTED** [review]

File: cob/cob.c (hunk 2 above)
Comment (ydb242) [reply cob/cob.c]: Done.

File: be/extmap.c [hunk 7]
```diff
@@ -589,6 +591,8 @@ M0_INTERNAL void m0_be_emap_merge(struct m0_be_emap_cursor *it,
 	if (rc == 0)
 		rc = emap_it_get(it) /* re-initialise cursor position */ ?:
 			update_next_segment(it, tx, delta, inserted);
+	else
+		M0_RC(rc);
```
[the remark this answers is not in the document]
Comment (ydb242) [reply be/extmap.c]: Done.

**Review by kanchan-chaudhari — APPROVED** [review]
Looks good.

**Review by shashank-parulekar — APPROVED** [review]

**Pull request merged**